File: cases/taint/kotlin/exception-catch-positive/ExceptionCatchPositive.kt
package dataflowbench

object ExceptionCatchPositive {
    class FlowException : RuntimeException() {
        var value: Int = 0
    }

    fun dfb_source(): Int { // DFB-SOURCE: exception-catch-input
        return 1
    }

    fun dfb_sink(value: Int) {} // DFB-SINK: exception-catch-sink

    fun run() {
        try {
            val flow = FlowException()
            flow.value = dfb_source()
            throw flow // DFB-WITNESS: exception-catch-throw
        } catch (caught: FlowException) {
            dfb_sink(caught.value)
        }
    }
}

File: cases/taint/scala/exception-catch-positive/ExceptionCatchPositive.scala
package dataflowbench

object ExceptionCatchPositive {
  class FlowException extends RuntimeException {
    var value: Int = 0
  }

  def dfb_source(): Int = { // DFB-SOURCE: exception-catch-input
    1
  }

  def dfb_sink(value: Int): Unit = {} // DFB-SINK: exception-catch-sink

  def run(): Unit = {
    try {
      val flow = new FlowException()
      flow.value = dfb_source()
      throw flow // DFB-WITNESS: exception-catch-throw
    } catch {
      case caught: FlowException => dfb_sink(caught.value)
    }
  }
}
